### PR TITLE
Ofmcc 3481 next term details

### PR DESCRIPTION
**Pull request by jenbeckett**
a lot going on in this PR 
https://eccbc.atlassian.net/browse/OFMCC-3973 - show when the funds expire on the Supp App Page

3481 - Update the missing info links to go back to the appropriate section. This opened a whole can of worms that the submit page did not handle having both a current and next year supp app. Most of the code in this PR addresses these issues. 

In order to link back to the correct current / next year component, I use a url query to set the appropriate data field on the previous screen. The downfall to this - is most of the time the 'nextYear=false' appears in the url bar. If you don't like this @weskubo-cgi I can try and implement it differently so it only shows in the URL if it's true

There was also no wireframes showing both current/ next year supp apps on the screen at the same time... So I kinda just went for it and updated the screens the best I could but I'm also open to suggestions for how the submit screen is. 

A small fix around the expand / collapse buttons where they would break after switching from current to next year. 

https://eccbc.atlassian.net/browse/OFMCC-4405

### CODE REVIEW

File: frontend/src/components/supp-allowances/IndigenousProgrammingSummary.vue [hunk 1]
```diff
@@ -1,55 +1,61 @@
...
   },
-  computed: {
-    selectedFunding() {
-      return this.INDIG_CHECKBOX_LABELS.filter((item) => this.indigenousProgrammingModel.indigenousFundingModel.includes(item.value))
-    },
-    isOtherBoxDisplayed() {
-      return this.indigenousProgrammingModel.indigenousFundingModel.includes(this.INDIG_CHECKBOX_LABELS.find((item) => item.label === 'Other').value)
-    },
-  },
+  computed: {},
```
Comment (vietle-cgi) [inline frontend/src/components/supp-allowances/IndigenousProgrammingSummary.vue:47]: we can remove the empty computed

File: frontend/src/components/supp-allowances/SupportNeedsSummary.vue [hunk 2]
```diff
@@ -1,55 +1,61 @@
...
   },
-  computed: {
-    selectedFunding() {
-      return this.SUPPORT_CHECKBOX_LABELS.filter((item) => this.supportModel.supportFundingModel.includes(item.value))
-    },
-    isOtherBoxDisplayed() {
-      return this.supportModel.supportFundingModel.includes(this.SUPPORT_CHECKBOX_LABELS.find((item) => item.label === 'Other').value)
-    },
-  },
+  computed: {},
```
Comment (vietle-cgi) [inline frontend/src/components/supp-allowances/SupportNeedsSummary.vue:47]: we can remove the empty computed

File: frontend/src/views/supp-allowances/SupplementaryFormView.vue [hunk 3]
```diff
@@ -7,21 +7,29 @@
...
       <v-col cols="6" lg="2" class="">
-        <AppButton id="next-term-button" :active="nextTermActive" :primary="false" :disabled="!isNextTermEnabled" size="large" width="200px" @click="setActiveTerm(true)">Next Term</AppButton>
+        <AppButton id="next-year-button" :active="nextTermActive" :primary="false" :disabled="!isNextTermEnabled" size="large" width="200px" @click="setActiveTerm(true)">Next Year</AppButton>
+      </v-col>
+    </v-row>
+    <v-row v-if="fundingExpiryDate">
+      <v-col cols="12" class="">
+        <div v-if="!nextTermActive">
+          If you apply for and receive funding in the current year of your funding agreement, the funds must be used by
+          {{ new Date(Date.parse(fundingExpiryDate.toISOString())).toLocaleString('en-US', { timeZone: 'UTC', dateStyle: 'full' }) }}
```
Comment (vietle-cgi) [inline frontend/src/views/supp-allowances/SupplementaryFormView.vue:31]: I think we can add this one to format.js, so we can reuse it in other components? also, we normally use locale 'en-CA' (although I think 'en-US' and 'en-CA')

**Review by vietle-cgi — CHANGES_REQUESTED** [review]

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: frontend/src/components/supp-allowances/IndigenousProgrammingAllowance.vue [hunk 4]
```diff
@@ -89,7 +89,7 @@ export default {
   },
   computed: {
     isOtherBoxDisplayed() {
-      return this.model.indigenousFundingModel.includes('9')
+      return this.model?.indigenousFundingModel?.includes('9')
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/components/supp-allowances/IndigenousProgrammingAllowance.vue:92]: done

File: frontend/src/components/supp-allowances/IndigenousProgrammingSummary.vue (hunk 1 above)
Comment (jenbeckett) [reply frontend/src/components/supp-allowances/IndigenousProgrammingSummary.vue:47]: done

File: frontend/src/components/supp-allowances/SupportNeedsSummary.vue (hunk 2 above)
Comment (jenbeckett) [reply frontend/src/components/supp-allowances/SupportNeedsSummary.vue:47]: done

File: frontend/src/views/supp-allowances/SupplementaryFormView.vue (hunk 3 above)
Comment (jenbeckett) [reply frontend/src/views/supp-allowances/SupplementaryFormView.vue:31]: yes good call moving it to format.js - and I also have changed it to use CA instead :)

File: frontend/src/views/supp-allowances/SupplementaryFormView.vue [hunk 5]
```diff
@@ -267,6 +277,7 @@ export default {
     this.DAYS_BEFORE_TERM_EXPIRES = 45
     this.DAYS_BEFORE_NEXT_TERM_ENABLED = 120
     this.setSuppTermDates()
+    this.route = useRoute()
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/supp-allowances/SupplementaryFormView.vue:280]: done

File: frontend/src/views/supp-allowances/SupplementaryFormView.vue [hunk 6]
```diff
@@ -7,21 +7,29 @@
   </p>
 
   <AppAlertBanner v-if="currentTermDisabled" type="warning">
-    <div>Your current term funding is ending and you are no longer able to make changes. Please apply for Next Term</div>
+    <div>Your current year funding is ending and you are no longer able to make changes. Please apply for Next Year</div>
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/supp-allowances/SupplementaryFormView.vue:10]: No I think it was left over from the old component. it has been removed

File: frontend/src/views/supp-allowances/SupplementaryFormView.vue [hunk 7]
```diff
@@ -7,21 +7,29 @@
...
-        <AppButton id="current-term-button" class="mr-1" :active="!nextTermActive" :primary="false" size="large" width="200px" @click="setActiveTerm(false)">Current Term</AppButton>
+        <AppButton id="current-year-button" class="mr-1" :active="!nextTermActive" :primary="false" size="large" width="200px" @click="setActiveTerm(false)">Current Year</AppButton>
       </v-col>
       <v-col cols="6" lg="2" class="">
-        <AppButton id="next-term-button" :active="nextTermActive" :primary="false" :disabled="!isNextTermEnabled" size="large" width="200px" @click="setActiveTerm(true)">Next Term</AppButton>
+        <AppButton id="next-year-button" :active="nextTermActive" :primary="false" :disabled="!isNextTermEnabled" size="large" width="200px" @click="setActiveTerm(true)">Next Year</AppButton>
+      </v-col>
+    </v-row>
+    <v-row v-if="fundingExpiryDate">
+      <v-col cols="12" class="">
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/supp-allowances/SupplementaryFormView.vue:28]: done

File: frontend/src/components/supp-allowances/TransportationSummary.vue [hunk 8]
```diff
@@ -4,22 +4,26 @@
...
       </v-row>
-      <v-divider class="my-3"></v-divider>
+      <v-divider class="mt-3"></v-divider>
 
-      <v-row v-if="model.VIN && model.odometer && model.estimatedMileage && !areDocumentsMissing(model) && !hasDuplicateVIN(model, allTransportModels)" no-gutters class="mt-2 pt-2">
+      <AppAlertBanner v-if="model.isNextTerm" type="info">
+        <div>This Supplementary Application is for Next Year</div>
+      </AppAlertBanner>
+
+      <v-row v-if="model.VIN && model.odometer && model.estimatedMileage && !areDocumentsMissing(model) && !hasDuplicateVIN(model, allTransportModels)" no-gutters class="mt-2 py-2">
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/components/supp-allowances/TransportationSummary.vue:21]: this logic has been moved to method

File: frontend/src/views/supp-allowances/SupplementaryFormView.vue [hunk 9]
```diff
@@ -94,7 +102,7 @@
         </v-expansion-panel>
       </v-expansion-panels>
     </div>
-    <div v-if="nextTermActive">
+    <div v-show="nextTermActive">
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/supp-allowances/SupplementaryFormView.vue:105]: fixed. I originally thought having the v-if was removing the data when switching back and forth quickly - but the issue was actually what UUID method I was calling.

File: frontend/src/views/supp-allowances/SupplementarySubmitView.vue [hunk 10]
```diff
@@ -258,8 +268,33 @@ export default {
...
       }
     },
-    getTransportModels() {
-      return this.models?.filter((el) => el.supplementaryType === SUPPLEMENTARY_TYPES.TRANSPORT)
+    setSuppTermDates() {
+      const today = new Date()
+
+      const formattedEndDate = new Date(this.fundingAgreement.endDate)
+      const termTwoEndDate = new Date(new Date(formattedEndDate).setFullYear(new Date(formattedEndDate).getFullYear() - 1))
+      const termOneEndDate = new Date(new Date(termTwoEndDate).setFullYear(new Date(termTwoEndDate).getFullYear() - 1))
```
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/supp-allowances/SupplementarySubmitView.vue:276]: yup! done

File: frontend/src/views/supp-allowances/SupplementaryFormView.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (jenbeckett) [reply frontend/src/views/supp-allowances/SupplementaryFormView.vue:31]: I moved it to format.js per the other comment but I also simplified as well

**Review by weskubo-cgi — APPROVED** [review]

**Review by vietle-cgi — APPROVED** [review]

**Pull request merged**